addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiBridgeHandler.java

```diff
@@ -0,0 +1,91 @@
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.nuki.NukiBindingConstants;
+import org.openhab.binding.nuki.dataexchange.BridgeInfoResponse;
+import org.openhab.binding.nuki.dataexchange.NukiHttpClient;
+import org.openhab.binding.nuki.dataexchange.NukiHttpServer;
+import org.openhab.binding.nuki.dataexchange.NukiHttpServerListener;
+import org.openhab.binding.nuki.dto.BridgeApiLockStateRequestDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NukiBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiBridgeHandler extends BaseBridgeHandler implements NukiHttpServerListener {
+
+    private final static Logger logger = LoggerFactory.getLogger(NukiBridgeHandler.class);
+
+    private NukiHttpServer nukiHttpServer;
+
+    public NukiBridgeHandler(Bridge bridge) {
+        super(bridge);
+        logger.trace("Instantiating NukiBridgeHandler({})", bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("NukiBridgeHandler:initialize()");
+        nukiHttpServer = NukiHttpServer.getInstance(this.getConfig(), this);
+        BridgeInfoResponse bridgeInfoResponse = new NukiHttpClient(this.getConfig()).getBridgeInfo();
+        if (bridgeInfoResponse.getStatus() == 200) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                    "Found " + bridgeInfoResponse.getBridgeInfo().getScanResults().size() + " Nuki Smart Locks.");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, bridgeInfoResponse.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiBridgeHandler:handleCommand({}, {})", channelUID, command);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiBridgeHandler:dispose");
+        nukiHttpServer.stopServer();
+    }
+
+    @Override
+    public void handleBridgeLockStateChange(BridgeApiLockStateRequestDto bridgeApiLockStateRequestDto) {
+        logger.debug("NukiBridgeHandler:handleBridgeLockStateChange({})", bridgeApiLockStateRequestDto);
+        String nukiId = String.valueOf(bridgeApiLockStateRequestDto.getNukiId());
+        Bridge bridge = this.getThing();
+        List<Thing> things = bridge.getThings();
+        for (Thing thing : things) {
+            String nukiIdThing = (String) thing.getConfiguration().get(NukiBindingConstants.CONFIG_NUKIID);
+            if (nukiId.equals(nukiIdThing)) {
+                Channel channel = thing.getChannel(NukiBindingConstants.CHANNEL_SMARTLOCKOPENCLOSE);
+                State state = bridgeApiLockStateRequestDto.getState() == 1 ? OnOffType.ON : OnOffType.OFF;
```

Comment:
I think in certain installs, Nuki can unlock the door by turning the key so far, and then pull the latch. A lot of the european stuff uses the key to unlock, which simply unblocks the multi point locking rods, then the door needs the handle pressed down to open the multi point locking rods. To lock the handle has to be lifted up from its resting position to lock the locking rods, and then the key can be turned to lock the system in place.
From what I've found out so far some mechanisms offer the 2 functions combined and some don't, hence the different options. Maybe this is selected in the app when the type of door handle is specified.
I seem to remember hearing about "pulling the latch" in one of their videos on the website although I don't know which type of mechanism I will get with my new door. My current sliding doors have the 2 functions separated.
Hope this makes sense!